site/content/integrate/apps/deployment/_index.md

```diff
@@ -52,32 +52,116 @@ For a go app the manifest snippet would look like this:
         }
     ]
 }
-``` 
+```
+
+### Setting up your Mattermost instance and AWS for deploying third-party apps
+
+Note that third-party apps are not supported in the Mattermost Cloud - they're used only for developer testing convenience. Provisioning in the third-party AWS cloud environment is done by the **appsctl** tool after completing the environment set up.
+
+We need an app bundle to provision an app. The bundle might be provisioned from the local disk, from S3 (not implemented yet), or from a URL (not implemented yet). Provisioning consists of three parts which will be completed via the **appsctl** tool:
+
+1. Creating the lambda functions with appropriate policies.
+2. Storing static assets in the dedicated S3 bucket.
+3. Storing the app’s manifest file in the same dedicated S3 bucket.
+
+#### Mattermost setup
+
+Follow the Prerequisites section of the [JS](../quick-start-js) or [GO](../quick-start-go) Quick Start Guides to set up your Mattermost instance. Ensure that you install the [apps plugin](https://developers.mattermost.com/integrate/apps/quick-start-js/#install-the-apps-plugin) as part of this process.
+
+#### AWS setup
+
+##### 1. Create an IAM user and access key and secret
```

Comment:
1/5 we should explain clearly that there are are 2 usually different sets of credentials involved, those used for provisioning (can be a privileged IAM user, or even the account owner's personal access key), and "runtime" used by the Mattermost instance to invoke Apps.

The ^^ is provisioning

Reply:
I also agree with this suggestion. I reordered the AWS Setup instructions.  I think it helps to move the `create s3 bucket` step first.  This is a bit of a time-consuming step and doesn't require any of the credentials the user will need.

step 2 (creating an IAM user) requires some work in AWS.

Now, steps 3-5 are all very simple steps that only set variables and run the init command. 

@DHaussermann, would you review this also? I think this makes the instructions flow much better.

**From** 
AWS setup
1. Create an IAM user and access key and secret
2. Create AWS S3 bucket
3. Set AWS environment variables

**To**
AWS setup
1. Create AWS S3 bucket
2. Create an IAM user access key and secret
3. Set AWS Provision environment variables
4. Initialize the AWS resources
5. Set AWS Invoke environment variables

[`78beeb6` (#826)](https://github.com/mattermost/mattermost-developer-documentation/pull/826/commits/78beeb6c99739cbe855636b7f59b01e0627d90db)